test/e2e/connectivity_test.go

```diff
@@ -135,10 +144,11 @@ func TestHostPortPodConnectivity(t *testing.T) {
 	data.testHostPortPodConnectivity(t)
 }
 
-// createPodsOnDifferentNodes creates numPods busybox test Pods and assign them to all the different
-// Nodes in round-robin fashion, then returns the names of the created Pods as well as a function
-// which will delete the Pods when called.
-func createPodsOnDifferentNodes(t *testing.T, data *TestData, numPods int) (podNames []string, cleanup func()) {
+// createPodsOnDifferentNodes creates numPods busybox test Pods and assign them to Nodes in the following fashion. If there
+// are any Windows Nodes, at least involve one. Then if there're > 1 Windows Nodes, involve another Windows one. Then
+// involve one Linux Node. Finally, involve the remaining Pods in the round-robin fashion. This function returns the names
+// of the created Pods as well as a function which will delete the Pods when called.
+func createPodsOnDifferentNodes(t *testing.T, data *TestData, numPods int) (podNames []string, isWindows []bool, cleanup func()) {
```

Comment:
I have been thinking about this and I think that it may be simpler to replace this function that creates a DaemonSet with a test Pod on each Node.
If we can switch to a different multi-arch image (agnhost?), the code would be much simpler.

Reply:
Sounds good. I am working on it.

Reply:
Updated.